src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -54,13 +55,24 @@ internal static Summary[] Run(BenchmarkRunInfo[] benchmarkRunInfos)
             using (var streamLogger = new StreamLogger(GetLogFileStreamWriter(benchmarkRunInfos, logFilePath)))
             {
                 var compositeLogger = CreateCompositeLogger(benchmarkRunInfos, streamLogger);
+                var eventHandler = CreateCompositeEventHandler(benchmarkRunInfos);
+
+                eventHandler.HandleStartValidationStage();
 
                 compositeLogger.WriteLineInfo("// Validating benchmarks:");
 
                 var (supportedBenchmarks, validationErrors) = GetSupportedBenchmarks(benchmarkRunInfos, resolver);
 
                 validationErrors.AddRange(Validate(supportedBenchmarks));
 
+                foreach (var validationError in validationErrors)
+                {
+                    var validationErrorEventHandler = validationError.BenchmarkCase != null
+                        ? validationError.BenchmarkCase.Config.GetCompositeEventHandler()
+                        : eventHandler;
+                    validationErrorEventHandler.HandleValidationError(validationError);
```

Comment:
I am not sure whether we should be allowing to have the event handlers only for selected benchmarks.

```suggestion
                    eventHandler.HandleValidationError(validationError);
```

@caaavik-msft what do you think? do you see any real life usage examples of that?

Reply:
The only case I can think of would be for the BenchmarkDotNet.Samples project where we might want to demonstrate a custom event handler but not apply it to every benchmark that is being run. But I suppose even for that case it is probably acceptable since usually only one type is run at a time when using the samples.